warp10/src/main/java/io/warp10/script/functions/MOTIONSPLIT.java

```diff
@@ -32,174 +33,303 @@
  */
 public class MOTIONSPLIT extends ElementOrListStackFunction {
 
+  public static final String PARAM_TIMESPLIT = "timesplit";
+  public static final String PARAM_START_LABEL = "label.split.start";
+  public static final String PARAM_PROXIMITY_IN_ZONE_TIME_LABEL = "label.stopped.time";
+  public static final String PARAM_SPLIT_TYPE_LABEL = "label.split.type";
+  public static final String PARAM_DISTANCETHRESHOLD = "distance.split";
+  public static final String PARAM_PROXIMITY_ZONE_TIME = "stopped.min.time";
+  public static final String PARAM_PROXIMITY_ZONE_SPEED = "stopped.max.speed";
+  public static final String PARAM_PROXIMITY_ZONE_RADIUS = "stopped.max.radius";
+  public static final String PARAM_PROXIMITY_IN_ZONE_MAX_SPEED = "stopped.max.mean.speed";
+
+  public static final String SPLIT_TYPE_TIME = "timesplit";
+  public static final String SPLIT_TYPE_DISTANCE = "distancesplit";
+  public static final String SPLIT_TYPE_END = "end";
+  public static final String SPLIT_TYPE_STOPPED = "stopped";
+
+
   public MOTIONSPLIT(String name) {
     super(name);
   }
-  
+
   @Override
   public ElementStackFunction generateFunction(WarpScriptStack stack) throws WarpScriptException {
+
+    Map<String, Object> params = null;
     Object top = stack.pop();
-    
-    if (!(top instanceof Number)) {
-      throw new WarpScriptException(getName() + " expects a proximity zone speed in m/s.");
+
+    if (!(top instanceof Map)) {
+      throw new WarpScriptException(getName() + " expects a parameter MAP on top of the stack.");
+    }
+
+    params = (Map) top;
+
+    long timeThreshold = Long.MAX_VALUE;
+    if (params.containsKey(PARAM_TIMESPLIT)) {
```

Comment:
You can simplify by doing o = params.get(XXX) and then throwing an Exception if o is not of the expected type, since you don't support null as a valid value for any of the parameters